pkg/cmdutil/userio/wizard.go

```diff
@@ -0,0 +1,247 @@
+	width           int
+}
+
+type task struct {
+	title          string
+	completedTitle string
+	completed      bool
+	logs           []logMsg
+}
+
+type updateCurrentTaskCompletedTitle string
+type logMsg struct {
+	message string
+	level   zerolog.Level
+}
+type doneMsg bool
+
+var checkMark = fmt.Sprint(lipgloss.NewStyle().Foreground(lipgloss.Color("42")).SetString("✓"))
+
+func (sm wizardModel) ReceiveUpdateMessages() tea.Msg {
+	message := <-sm.messageChan
+	return message
+}
+
+func (sm wizardModel) Init() tea.Cmd {
+	return tea.Batch(
+		sm.model.Tick,
+		sm.ReceiveUpdateMessages,
+	)
+}
+
+func (sm wizardModel) Update(msg tea.Msg) (tea.Model, tea.Cmd) {
+	if _, ok := msg.(spinner.TickMsg); !ok {
+		if _, ok := msg.(tea.Model); ok {
+			log.Debug().Msgf("Wizard: Received msg [%T]", msg)
+		} else {
+			log.Debug().Msgf("Wizard: Received msg [%T] %s", msg, msg)
+		}
+	}
+
+	updateListener := sm.ReceiveUpdateMessages
+
+	switch msg := msg.(type) {
+	case tea.WindowSizeMsg:
+		sm.width = msg.Width
+		sm.height = msg.Height
+		if sm.inputModel != nil {
+			newInputModel, inputCmd := sm.inputModel.Update(msg)
+			sm.inputModel = newInputModel
+			return sm, inputCmd
+		}
+		return sm, nil
+	case doneMsg:
+		if len(sm.messageChan) > 0 {
+			return sm, updateListener
+		} else {
+			// Mark previous tasks as complete if we are done
+			if len(sm.tasks) > 0 {
+				sm.tasks[len(sm.tasks)-1].completed = true
+			}
+			sm.quitting = true
+			return sm, tea.Quit
+		}
+	case task:
+		// Mark previous tasks as complete if we add another
+		if len(sm.tasks) > 0 {
+			sm.tasks[len(sm.tasks)-1].completed = true
+		}
+		sm.tasks = append(sm.tasks, msg)
+		return sm, updateListener
+	case logMsg:
+		if len(sm.tasks) > 0 {
+			// Adds logs as children of the most recent task
+			sm.tasks[len(sm.tasks)-1].logs = append(sm.tasks[len(sm.tasks)-1].logs, msg)
+		} else {
+			log.Warn().Msgf("Could not add log, no active tasks [%s: %s]", msg.level, msg.message)
+		}
+		return sm, updateListener
+	case updateCurrentTaskCompletedTitle:
+		if len(sm.tasks) > 0 {
+			sm.tasks[len(sm.tasks)-1].completedTitle = string(msg)
+		}
+		return sm, updateListener
+	case InputCompleted:
+		sm.inputResultChan <- sm.inputModel
+		sm.inputModel = nil
+		return sm, updateListener
+	case tea.Model:
+		var cmd tea.Cmd
+		sm.inputModel, cmd = msg.Update(tea.WindowSizeMsg{Width: sm.width, Height: sm.height})
+		return sm, tea.Sequence(updateListener, cmd)
+	case tea.KeyMsg:
+		switch msg.Type {
+		case tea.KeyCtrlC:
+			sm.err = ErrInterrupted
+			sm.quitting = true
+			return sm, tea.Quit
+		}
+	}
+	var newInputModel tea.Model
+	var inputCmd tea.Cmd
+	if sm.inputModel != nil {
+		newInputModel, inputCmd = sm.inputModel.Update(msg)
+		sm.inputModel = newInputModel
+	}
+	newSpinnerModel, cmd := sm.model.Update(msg)
+	sm.model = newSpinnerModel
+	return sm, tea.Batch(cmd, inputCmd)
+}
+
+func (sm wizardModel) View() string {
+	bold := lipgloss.NewStyle().Bold(true)
+	buffer := ""
```

Comment:
It's better to use something like `strings.Builder` for creating dynamic strings like here. The current solution copies `buffer` every `+=` operation